src/js/edu-benefits/5490/helpers.jsx

```diff
@@ -1,8 +1,23 @@
+import _ from 'lodash/fp';
+
 import React from 'react';
 import { transformForSubmit } from '../../common/schemaform/helpers';
 
 export function transform(formConfig, form) {
-  const formData = transformForSubmit(formConfig, form);
+  // Clone the form in so we don't modify the original...because of reasons FP
+  const newForm = _.cloneDeep(form);
+
+  // Copy the data if necessary
+  if (form.sponsorInformation.data['view:currentSameAsPrevious']) {
+    newForm.sponsorInformation.data['view:currentSponsorInformation'] = {
+      veteranFullName: form.benefitHistory.data.previousBenefits.veteranFullName,
+      'view:veteranId': {
+        veteranSocialSecurityNumber: form.benefitHistory.data.previousBenefits.veteranSocialSecurityNumber
```

Comment:
I'm now surprised that for previouesBenefits, we're not allowing either the SSN or VA File Number, but that's another story for another day/PR. (I think we should ask stakeholders about that--would be a better experience to be consistent about what info is required about any sponsor.)